jmeter/README.md

```diff
@@ -28,29 +28,61 @@ The test plan can be ran straight from the command line.  A helper script is pro
 ```
 
+_Run the test plan in query mode 80 total times in 10 threads each completing 8 loops._
 ```sh
-./run-tests.sh fetch 10 8 my-s3-bucket
+./run-tests.sh query 10 8 my-project-name my-s3-bucket
 ```
 
+_Run in batch mode.  Note that all feeds in the csv file will be processed in each loop.  So in the follow command, each feed in the batch.csv file would be processed 6 times._
+```sh
+./run-tests.sh query 3 2 batch.csv my-s3-bucket
+```
+
+### Running the upload test on multiple gtfs files
+
+As noted above, the jmeter script can be run in `batch` mode.  The provded csv file must contain the following headers and data:
+
+| header | description |
+| ---- | ---- |
+| project name | name of project to be created |
+| mode | Must be either `fetch` or `upload` |
+| location | The path to the file if the mode is `upload` or the http address if the mode is `fetch` |
+
+There is also a helper python script that can be used to run the jmeter script in `batch` mode using all files stored within an s3 bucket.  This script requires that aws credentials have been setup for use by the aws command line tool.
+
+| # | argument | possible values | description |
+| ---- | ---- | ---- | ---- |
+| 1 | test plan mode | `fetch` or `upload` | The test plan mode to use.  This will be written to each row of the csv file described above. |
+| 2 | s3 bucket of gtfs feeds  | the string of an s3 bucket | An s3 bucket that is accessbile with the credentials setup for the aws cli.  Place zip files within the bucket.  Each zip file will be downloaded to the local machine and the jmeter test plan will be ran in upload mode for each gtfs zip file. |
+| 3 | s3 bucket for output reports | the string of an s3 bucket | OPTIONAL. After each test run, the script will tar up the output folder and attempt to upload to the specified s3 bucket.  |
+
+Example:
+
+```sh
+python run-upload-tests.py fetch gtfs-test-feeds datatools-jmeter-results
+```
+
+
 ## Test Plan
 
-A single test plan file is used for maintainablility.  By default, the test plan runs 1 thread in 1 loop and will upload a feed and then perform various checks on the uploaded feed version.  As noted in the above section, it is possible to run a different variation of the test plan that will not upload a feed and instead select a random project and then a random feed source from the project and then a random feed version from the feed source and then perform the same checks on the feed version.
+A single test plan file is used for maintainablility.  By default, the test plan runs 1 thread in 1 loop and will upload a feed and then perform various checks on the uploaded feed version.  As noted in the above section, it is possible to run different variations of the test plan.  There are 3 types of test plans that can be initiated: `fetch`, `query` or `upload`.  Also, there is a mode called `batch` that will read in a csv file of feeds that can be ran through either the `upload` or `fetch` test plans.
```

Comment:
Even though it will be short, I would still break out the description of the `batch` mode into its own heading (or perhaps just linking to the "Running the upload test on multiple gtfs files" section).